crypto/evp/evp_lib.c

```diff
@@ -114,6 +120,64 @@ int EVP_CIPHER_set_asn1_iv(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
     return i;
 }
 
+int EVP_CIPHER_get_asn1_aead_params(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
+{
+    int i = 0;
+    long tl;
+    unsigned char *iv;
+
+    if (type == NULL) {
+        return i;
+    }
+
+    if (!(c->cipher->flags & EVP_CIPH_FLAG_AEAD_CIPHER)) {
+        return -1;
+    }
+
+    i = ASN1_TYPE_get_octetstring_int(type, &tl, NULL, INT_MAX);
+    if (i <= 0 || (iv = OPENSSL_malloc(i)) == NULL)
+        return -1;
+    ASN1_TYPE_get_octetstring_int(type, &tl, iv, i);
```

Comment:
Why do we do nothing with the taglen here?

Reply:
It's not really needed as it should be the same as `i`. That param is not really useful. I just added there to be consistent with `ASN1_TYPE_get_int_octetstring`.